leonardo-lfonzi62/week-6/react-giphy-Hw Submission/src/components /search.js

```diff
@@ -0,0 +1,66 @@
+import React, { Component } from 'react';
+import Results from './results';
+import axios from 'axios';
+
+class Search extends Component {
+    state = {
+        Query:"",
+        response:[]
+    }
+
+    handleSubmit = (event) => {
+    event.preventDefault();
+    // making AJAX call to giphy database using the necessary requirements per the website instructions 
+        axios.get(`http://api.giphy.com/v1/gifs/search?q=${this.state.Query}&api_key=55wBK7VLZy5IoUOK0dFBERd0vHRdaN8A`)
+        .then((res) => {    
+            this.setState({
+                response: res.data.data,  // once the giphy data is received, updating the state of the empty response array to the giphy search data
+                Query: "" // reseting the state of Query to an empty string for UI improvements
+            });
+        })
+    
+    let QueryTerm =this.state.Query; 
+    console.info(`A new Query was submitted:${QueryTerm}`) // logging the user's search term input in the console 
+   
+}
+
+     handleChange =(event) => {
+    console.log('Your Query Term has changed!:' + event.target.value); // logging the Query term the user types in
+    this.setState({Query: event.target.value}); //dynamically  setting the state of the user's Query to whatever they are typing in the input
+};
+    
+    render() {
+
+        let gifresults;
+        if(this.state.response){ 
+            gifresults = this.state.response.map((dataResults) => {
+            //setting the gifresults var to equal to the gif data sent back from the API and mapping over that array
```

Comment:
Alot of great comments. I can see you understand how this data is flowing. I would however in your return minimize the amount of jsx is there. Since section and div are included for every gif I would move that code into the results component. Your function will be cleaner with that change.